src/main/java/org/openstreetmap/atlas/checks/validation/intersections/LineCrossingWaterBodyCheck.java

```diff
@@ -424,16 +437,19 @@ private boolean canCrossWaterBody(final AtlasItem crossingItem,
      */
     private Set<Tuple<PolyLine, Set<Location>>> getInteractionsPerWaterbodyComponent(
             final GeometricSurface waterbody, final AtlasObject object,
-            final PolyLine intersectingFeature)
+            final PolyLine intersectingFeature, final CheckFlag flag)
     {
         if (waterbody instanceof Polygon)
         {
-            final Set<Location> intersectionLocations = ((Polygon) waterbody)
+            Set<Location> intersectionLocations = ((Polygon) waterbody)
                     .intersections(intersectingFeature);
             if (intersectionLocations.isEmpty())
             {
                 return Set.of(Tuple.createTuple((Polygon) waterbody, Set.of()));
             }
+            intersectionLocations = Arrays.stream(intersectionLocations.toArray(new Location[0]))
```

Comment:
I am confused about what this line is doing. It looks like it is taking a set, making it into an array, streaming it, and then making it back into a set without really changing anything. Am I missing something?

Reply:
yeah I just forgot to come back to this and take it out.